pkg/datapath/linux/ipsec.go

```diff
@@ -213,24 +217,24 @@ func (n *linuxNodeHandler) enableIPsecIPv4(newNode *nodeTypes.Node, nodeID uint1
 			}
 
 			for _, cidr := range n.nodeConfig.IPv4PodSubnets {
-				spi, err = ipsec.UpsertIPsecEndpoint(wildcardCIDR, cidr, localIP, remoteIP, nodeID, newNode.BootID, ipsec.IPSecDirOut, zeroMark, updateExisting)
+				spi, err = ipsec.UpsertIPsecEndpoint(wildcardCIDR, cidr, localIP, remoteIP, nodeID, newNode.BootID, ipsec.IPSecDirOut, zeroMark, updateExisting, ipsec.DefaultReqID)
 				errs = errors.Join(errs, upsertIPsecLog(err, "out IPv4", wildcardCIDR, cidr, spi, nodeID))
 				if err != nil {
 					statesUpdated = false
 				}
 
 				/* Insert wildcard policy rules for traffic skipping back through host */
-				if err = ipsec.IpSecReplacePolicyFwd(cidr, localIP); err != nil {
+				if err = ipsec.IpSecReplacePolicyFwd(cidr, localIP, ipsec.DefaultReqID); err != nil {
 					log.WithError(err).Warning("egress unable to replace policy fwd:")
 				}
 
-				spi, err := ipsec.UpsertIPsecEndpoint(wildcardCIDR, cidr, localCiliumInternalIP, remoteCiliumInternalIP, nodeID, newNode.BootID, ipsec.IPSecDirIn, zeroMark, updateExisting)
+				spi, err := ipsec.UpsertIPsecEndpoint(wildcardCIDR, cidr, localCiliumInternalIP, remoteCiliumInternalIP, nodeID, newNode.BootID, ipsec.IPSecDirIn, zeroMark, updateExisting, ipsec.DefaultReqID)
```

Comment:
Do you mean shadow `spi` and `err` inside the loop?

Reply:
Both err and spi are used immediately for logging purpose. We do update `errs` if err is not nil. `upsertIPsecLog` returns err or logs debug message. I don't think there is shadowing concern, since `errs` will contain all returned errors. 

Usage of both SPI and err is not changed in this PR. Modification is adding additional parameter to `UpsertIPsecEndpoint`

Reply:
I'm probably the one who messed this up at some point :grimacing: No impact I can see, as Pabla said, but I'll try to send a fix anyway.

Reply:
@pchaigno I can fix it in this PR as well, seems straight forward change.

Reply:
Unless we want to backport that fix for some reason, then it is probably not a right PR to do it.

Reply:
I don't think we need to backport given the lack of impact, so feel free to make it in this PR. Thanks! :pray: